Skip check on temp. pods like ceph-file-controller-detect-version-jdmd4 for test_add_capacity[10]

ODF 4.17 failure- 

E               ocs_ci.ocs.exceptions.CommandFailed: Error during execution of command: oc --kubeconfig /home/jenkins/current-cluster-dir/openshift-cluster-dir/auth/kubeconfig -n openshift-storage get Pod ceph-file-controller-detect-version-jdmd4 -n openshift-storage.
E               Error is Error from server (NotFound): pods "ceph-file-controller-detect-version-jdmd4" not found

In code:
  restart_count_after = pod_helpers.get_pod_restarts_count(
            config.ENV_DATA["cluster_namespace"]
        )

--- a/ocs_ci/ocs/resources/pod.py
+++ b/ocs_ci/ocs/resources/pod.py
@@ -2333,7 +2333,9 @@ def get_pod_restarts_count(
     restart_dict = {}
     ocp_pod_obj = OCP(kind=constants.POD, namespace=namespace)
     for p in list_of_pods:
-        # we don't want to compare osd-prepare and canary pods as they get created freshly when an osd need to be added.
+        # we don't want to compare osd-prepare and canary pods as they get
+        # created freshly when an osd need to be added. Also skip check on ceph-file-controller-detect-version
+        # as it's a temp. pod.
         if (
             "rook-ceph-osd-prepare" not in p.name
             and "rook-ceph-drain-canary" not in p.name

--- a/ocs_ci/ocs/resources/pod.py
+++ b/ocs_ci/ocs/resources/pod.py
@@ -2337,6 +2337,7 @@ def get_pod_restarts_count(
         if (
             "rook-ceph-osd-prepare" not in p.name
             and "rook-ceph-drain-canary" not in p.name
+            and "ceph-file-controller-detect-version" not in p.name
         ):
             pod_count = ocp_pod_obj.get_resource(p.name, "RESTARTS")
             restart_dict[p.name] = int(pod_count.split()[0])

--- a/ocs_ci/ocs/resources/pod.py
+++ b/ocs_ci/ocs/resources/pod.py
@@ -2336,11 +2336,12 @@ def get_pod_restarts_count(
         # we don't want to compare osd-prepare and canary pods as they get
         # created freshly when an osd need to be added. Also skip check on ceph-file-controller-detect-version
         # as it's a temp. pod.
-        if (
-            "rook-ceph-osd-prepare" not in p.name
-            and "rook-ceph-drain-canary" not in p.name
-            and "ceph-file-controller-detect-version" not in p.name
-        ):
+        exclude_names = (
+            "rook-ceph-osd-prepare",
+            "rook-ceph-drain-canary",
+            "ceph-file-controller-detect-version",
+        )
+        if all(exclude_name not in p.name for exclude_name in exclude_names):
             pod_count = ocp_pod_obj.get_resource(p.name, "RESTARTS")
             restart_dict[p.name] = int(pod_count.split()[0])
     logger.info(f"get_pod_restarts_count: restarts dict = {restart_dict}")

